python/ray/_private/gpu_object_manager.py

```diff
@@ -141,12 +149,38 @@ def _recv_gpu_object(
             util.__ray_recv__, communicator_name, obj_id, src_rank, tensor_meta
         )
 
+    def fetch_gpu_object(self, obj_id: str):
+        """
+        Fetches the GPU object from the source actor's GPU object store via the object store
+        instead of out-of-band tensor transfer and stores the tensors in `gpu_object_store`.
+
+        This is useful when the current process is the driver process that coordinates the data
+        transfer of this GPU object. For example, when the driver process calls `ray.get(gpu_object_ref)`,
+        the tensors will not be transferred out-of-band automatically. We need to fetch the tensors from
+        the corresponding actor's GPU object store via the object store API.
```

Comment:
```suggestion
        This is useful when the current process does not support the designated out-of-band tensor transport. For example, if the tensor transport is NCCL, but the driver does not have a GPU, we use this call to fulfill a `ray.get`.
```

Edited to be less specific to who the caller process should be.